cicerone/app/src/main/java/com/teampower/cicerone/GeofencingController.kt

```diff
@@ -0,0 +1,71 @@
+package com.teampower.cicerone
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.google.android.gms.location.Geofence
+import com.google.android.gms.location.GeofencingClient
+import com.google.android.gms.location.GeofencingRequest
+import com.google.android.gms.location.LocationServices
+
+class GeofencingController() {
+    private lateinit var geofencingClient: GeofencingClient
+    private var geofenceList = ArrayList<Geofence>()
+
+    fun startGeofencing(context: Context) {
+        val geofencePendingIntent: PendingIntent by lazy {
```

Comment:
What does this `by lazy` do for the intent?

Reply:
It was part of the guide, but lazy in general seems to make it so that the property is only initialized on first access. I.e. we don't have to initialize geofencinPendingIntent ourselves, it becomes initialized by the lambda function.